src/ServiceControl.Audit/Auditing/AuditIngestion.cs

```diff
@@ -187,8 +188,7 @@ async Task OnMessage(MessageContext messageContext, CancellationToken cancellati
                 await channel.Writer.WriteAsync(messageContext, cancellationToken);
                 await taskCompletionSource.Task;
 
-                ingestedMessagesCounter.Add(1);
-                messageSize.Record(messageContext.Body.Length / 1024.0);
+                successfulMessagesCounter.Add(1, tags);
             }
```

Comment:
I noticed in core that it also sets a tag for when the returned task is cancelled. Only issue here is that I think core does so incorrectly.

It should only add that tag when the cancellationtoken was set AND the task result is cancelled.

Not sure about how usable it but we could do:

```
finally
{
    if(cancellationToken.IsCancellationRequestion)
    tags.Add("cancelled", true);
}
```

Just before we do the `.Add`